config/decoder.go

```diff
@@ -231,6 +224,23 @@ func (d *decoder) mapping(childKey string, value reflect.Value, def string) erro
 	return nil
 }
 
+// Sets value to an interface type.
+func (d *decoder) iface(key string, value reflect.Value, def string) error {
+	v := d.getGlobalProvider().Get(key)
+
+	if !v.HasValue() || v.Value() == nil {
+		return nil
+	}
+
+	src := reflect.ValueOf(v.Value())
+	if src.Type().Implements(value.Type()) {
+		value.Set(src)
+		return nil
+	}
+
```

Comment:
@peter-edge: what's the argument against the newlines? most of the idiomatic code I've seen here, in other uber teams, and in the stdlib uses them to make the code more parsable at a glance. is there really a downside to them? just that the files get long?